### ISSUE TITLE
Terragrunt does not allow new `-chdir` option

### ISSUE THREAD

**Issue by zachwhaley** [gruntwork-io/terragrunt#1466]
Terraform 0.14 introduced a new global variable `-chdir` to allow Terraform to change working directory before executing.

https://www.terraform.io/docs/commands/index.html#switching-working-directory-with-chdir

Terragrunt does not seem to allow this option.

```shell
$ terragrunt -chdir=dir plan
Incorrect Usage. flag provided but not defined: -chdir
```

**Comment by lorengordon** [comment]
Pretty sure terragrunt requires the first argument to be the terraform command... Did you try `terragrunt plan -chdir=dir`?

**Comment by zachwhaley** [comment]
> Did you try `terragrunt plan -chdir=dir`?

Yes, this is an illegal Terrraform command since `-chdir` is a global option and must be put before the command.

```shell
$ terragrunt plan -chdir=dir
...
[terragrunt] 2020/12/15 12:29:46 Running command: terraform plan -out=tfplan -chdir=dir
Usage: terraform plan [options] [DIR]
...
[terragrunt] 2020/12/15 12:29:46 Hit multiple errors:
exit status 1
```

**Comment by yorinasub17** [comment]
Ah yup terragrunt doesn't support global flags for terraform. This probably needs a little bit of investigation to see where/how terraform global flags should be managed.

**Comment by sheldonhull** [comment]
This might be a pretty significant issue with `0.15` and on. The path being appended at the end is removed [Terraform Release Notes](https://www.terraform.io/upgrade-guides/0-15.html#commands-accepting-a-configuration-directory-argument)

For flexibility, chdir is required now to set context for the working directory.
This might also have some implications on the path scoping for the functions in terragrunt.

**Comment by lorengordon** [comment]
I'd be interested more in the use case this addresses, instead of using the option `--terragrunt-working-dir`, which seems to basically do the same thing?

**Comment by sheldonhull** [comment]
@lorengordon that's great to know about, so I'll look at using that. However, to be clear, the main concern I had is that the current syntax provided by terragrunt debug output shows to reproduce the run using `terraform  -var-file=foo directorypath`. This syntax is not supported in `0.15`. If terragrunt is not depending on this in the actual combination of arguments then this is just a debug message to update I suppose. I'll have to look at the source to know how that's being set. 

If it's part of the arguments passed to the terraform executable instead of changing directories directory, then it's a breaking change by Terraform. 

If it's just a logging message then it's just a new debug output for 0.15+ to provide chdir to reproduce a command instead.

**Comment by lorengordon** [comment]
> the current syntax provided by terragrunt debug output shows to reproduce the run using `terraform -var-file=foo directorypath`

I believe that would be an artifact of how you are running terragrunt. Instead of:

```
terragrunt plan directorypath
```

use:

```
terragrunt plan --terragrunt-working-dir directorypath
```

**Comment by sheldonhull** [comment]
Related to: #1651 

I just ran upgrade and am on version `0.29.0` and it failed with:
`Too many command line arguments. Did you mean to use -chdir?`

**Comment by sheldonhull** [comment]
@brikis98 wanted to add you as I tried the new release from PR #1651  and first thing I got was an error related to this same underlying change in chdir being used. This new subcommand syntax deprecates the ability to run `terraform plan {DIR}` as a path. I'm not using this syntax anyway, as I'm using the recommended `--terragrunt-working-dir`, but it still is failing with this. 

I tried looking for where the terraform argument construction was to see if I could identify where the syntax for directory is used, but I couldn't find it. Figured you might point me in the right direction and maybe if you are experiencing the same error since you started that PR.

**Comment by brikis98** [comment]
@sheldonhull Could you share your Terragrunt, config, what commands you're running, and the full log output so I can better understand the error you're seeing?

**Comment by malcolm061990** [comment]
@zachwhaley 
As mentioned before terragrunt has [terragrunt-working-dir](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-working-dir) option that does the same thing as `-chdir` in terraform.
I think we can close this issue :)

**Comment by jcmcken** [comment]
One thing that argues for supporting ``-chdir`` -- Terragrunt offers to be a thin Terraform wrapper, but doesn't support this commonly used flag and instead has its own, competing flag. There are some CI integrations with Terraform that use ``-chdir`` where we can't just drop Terragrunt in because of this issue.
